ui/web.go

```diff
@@ -72,8 +73,16 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
 	))
 
 	r.Post("/-/reload", func(w http.ResponseWriter, req *http.Request) {
-		w.Write([]byte("Reloading configuration file..."))
-		reloadCh <- struct{}{}
+		errc := make(chan error)
+		defer close(errc)
+
+		reloadCh <- errc
+		if err := <-errc; err != nil {
+			http.Error(w, fmt.Sprintf("config file reload failed: %v", err), http.StatusInternalServerError)
```

Comment:
not sure if it's necessary to provide the error via http, or it's enough to just say the reload failed. Open to opinions.